gh-128039: Clarify the uninstallation instructions for Mac. (#128040)
python/cpython: Doc/using/mac.rst

A default install will include:

* A |usemac_applications_folder_name| folder in your :file:`Applications` folder;

> * Within the |usemac_applications_folder_name| folder: :program:`IDLE`, the development environment that is a standard part of official Python distributions;

Reviewer: ```suggestion
    * Within the |usemac_applications_folder_name| folder: :program:`IDLE`, the development environment that is a standard part of official Python distributions;
```

Maybe this should be made a sub bullet point rather than standalone as it refers to the previous one?

> * Also within the |usemac_applications_folder_name| folder: :program:`Python Launcher`, which handles double-clicking

Reviewer: ```suggestion
    * Also within the |usemac_applications_folder_name| folder: :program:`Python Launcher`, which handles double-clicking
```
Same idea with this one

  Python scripts from the macOS `Finder <https://support.apple.com/en-us/HT201732>`_;

> * A :file:`3.13` folder (under the "framework" :file:`/Library/Frameworks/Python.framework`), which includes the

Reviewer 2: Rather than hardcoding `3.13` which is only correct for a single branch, let's use a "replacement":

```suggestion
* A |usemac_x_dot_y_literal| folder (under the "framework" :file:`/Library/Frameworks/Python.framework`), which includes the
```

And add it with the others in this file:

```diff
 .. |usemac_x_dot_y| replace:: 3.13
+.. |usemac_x_dot_y_literal| replace:: ``3.13``
 .. |usemac_python_x_dot_y_literal| replace:: ``python3.13``
 .. |usemac_python_x_dot_y_t_literal| replace:: ``python3.13t``
 .. |usemac_python_x_dot_y_t_literal_config| replace:: ``python3.13t-config``
```

And actually, those are still hardcoded, so let's remove the `usemac_` prefix and integrate them into the `rst_epilog` which uses the _current_ version number:

https://github.com/python/cpython/blob/be8ae086874cac42eea104c1ff21ef5868d50bdd/Doc/conf.py#L77-L80

The `rst_epilog` replacements are available for all pages.

Author: This is surely The Right Thing®, but I'm unfamiliar enough with ReStructured Text to do it myself (I took a stab at it and found myself going in circles after an hour).

Reviewer 2: That's fine, we can do it in a followup. Thank you for trying!

  Python executable and libraries (the installer adds this location to your shell
  path);

> 

Reviewer 3: There should also be a similar item for the free-threaded option, something like: "If you selected the Free-threaded Python option, there will also be a 3.x folder under /Library/Frameworks/PythonT.framework ..."

* Symlinks to the above Python executable, placed in :file:`/usr/local/bin/`.

> To uninstall Python, you can remove the above five things.

Reviewer 3: six
